drivers/video/video_display_legacy.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2019, Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * video_display_legacy is a shim layer allowing to expose a display compatible
+ * interface for video (display/output) devices. The video device needs to
+ * register display2video* methods as display_drv_api callbacks.
+ */
+
+#include <zephyr.h>
+
+#include <drivers/video.h>
+#include <drivers/display.h>
+
+#define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(video_display_legacy);
+
+/* Only one double-buffered display instance supported for now */
+static struct video_buffer *vbufa = NULL, *vbufb = NULL;
+
+int display2video_write(const struct device *dev, const u16_t x, const u16_t y,
+			const struct display_buffer_descriptor *desc,
+			const void *buf)
+{
+	struct video_buffer *vbuf;
+	struct video_format fmt;
+	int err, bpp, loop;
+	const u8_t *src;
+	u8_t *dst;
+
+	video_get_format(dev, VIDEO_EP_IN, &fmt);
+
+	/* retrieve available buffer */
+	err = video_dequeue(dev, VIDEO_EP_IN, &vbuf, K_FOREVER);
+	if (err) {
+		return err;
+	}
+
+	/* LVGL no double buffering aware support ? just copy previous one */
```

Comment:
Bit off topic but changes could be made to the LVGL glue logic to directly support the video API

Reply:
yes, I put that on the TODO list